map.js

```diff
@@ -195,143 +183,97 @@ function drawMapOverlay(){
     }
 }
 
-function overlayClick(lastMouseLoc){
+/**
+ * Handle click on the overlay layer.
+ * @param {Point} lastMouseLoc screen space coordinates of mouse click
+ * @returns if click was handled (ie, something was clicked on)
+ */
+function overlayClick(clickLoc){
+    //overlay coordinates are all in map space, so we convert to that before checking.
+    const clickLocInMapSpace = screenSpaceToMapSpace(clickLoc);
     if(currentOverlay == "Locations"){
         for(const icon of overlay.locations){
```

Comment:
This code could go in to an icon.onClick() method. As we add more features, we can decide if it is a good idea to move the code there, or to just keep it in this global method.

Reply:
I would keep this as a global method.

Stacking on another comment i made earlier, if we separate out the map topbar from map.js, the only clicking on the map would be either dragging/moving the map, or clicking on a map icon. (or am i forgetting something? >.<)